OJS.Workers.ExecutionStrategies/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs

```diff
@@ -159,9 +159,45 @@ class _$SandboxSecurityManager extends SecurityManager {
     }
 }";
 
-        public override ExecutionResult Execute(ExecutionContext executionContext)
+        protected static void UpdateExecutionTime(
+            string timeMeasurementFilePath,
+            ProcessExecutionResult processExecutionResult,
+            int timeLimit,
+            int updateTimeOffset)
+        {
+            if (File.Exists(timeMeasurementFilePath))
+            {
+                var timeMeasurementFileContent = File.ReadAllText(timeMeasurementFilePath);
+                if (long.TryParse(timeMeasurementFileContent, out var timeInNanoseconds))
+                {
+                    var totalTimeUsed = TimeSpan.FromMilliseconds((double)timeInNanoseconds / 1000000);
```

Comment:
Real magic DO exists :D (`1000000`)

Extracting the value into a `double` constant, will solve the division problem as well.